lib/active_model/serializer/adapter/json_api/link.rb

```diff
@@ -3,30 +3,43 @@ class Serializer
     module Adapter
       class JsonApi
         class Link
-          def initialize(serializer)
+          def initialize(serializer, value)
             @object = serializer.object
             @scope = serializer.scope
+            if value.respond_to?(:call)
+              self._value = instance_eval(&value)
+            else
+              self._value = value
```

Comment:
We should also probably comment that in the block form, if `meta` and/or `href` are used, the `Link` is an object.  If neither are used, the `Link` is a URI (the return value of the block).  If both are used, only the return value?